TTHAnalysis/python/tools/higgsRecoTTH.py

```diff
@@ -192,97 +412,215 @@ def run(self,event,Collection):
+                    #if (len(NuFromWFromH)>1):
+                        #print("WARNING! we have not one but " + str(len(NuFromWFromH)) + " neutrinos from W from H. I am in reconstruction loop. QFromWFromH is " + str(len(QFromWFromH)))
+                    for neutrino in NuFromWFromH:
+                        VisPlusNu = Hvisconstr+neutrino.p4() 
+                        pTVisPlusNu = VisPlusNu.Pt()
                     if mHvisconstr<self.cuts_mH_vis[0] or mHvisconstr>self.cuts_mH_vis[1]: continue
                     mindR = min(lep.DeltaR(j1),lep.DeltaR(j2))
                     delR_H_j1j2 = j1.DeltaR(j2)
-                    candidates.append((mindR,delR_H_j1j2,mHvisconstr,mW,_lep,_j1,_j2,pTHvisconstr))
-                    # need to remove #TODO
-                    # --------------
-                    #for jet in gengoodJets:
-                        #if deltaR(jet.p4().Eta(),jet.p4().Phi(), j1.Eta(),j1.Phi()) < 0.3 or deltaR(jet.p4().Eta(),jet.p4().Phi(), j2.Eta(),j2.Phi()) < 0.3:
-                           #print "at least one the detector-level jets matched with a true one --> counting it"
-                           #matchedjets +=1
-                        #elif deltaR(jet.p4().Eta(),jet.p4().Phi(), j1.Eta(),j1.Phi()) < 0.3 and deltaR(jet.p4().Eta(),jet.p4().Phi(), j2.Eta(),j2.Phi()) < 0.3:
-                             #print "both detector level jets match with both true ones --> counting it"
-                             #bothmatchedjets +=1
+                    candidateMinimizers.append((mindR,abs(mHvisconstr-125.0),abs(mW-80.4),delR_H_j1j2,_lep,_j1,_j2,pTVisPlusNu,pTHvisconstr))
+                    candidateBranchValues.append((mindR,mHvisconstr,mW,delR_H_j1j2,_lep,_j1,_j2,pTVisPlusNu,pTHvisconstr))
+            
             if self.useTopTagger:
                 for topjet in jetsTopNoB:
                     for gentopquark in QFromWFromT:
                         if topjet.p4().DeltaR(gentopquark.p4()) > 0.5:
                             #jets tagged as coming from top didn't match with true partons coming from top"
-                            nmismatchedtoptaggedjets +=1 #only with respect to the hadronic top where the W is going to qq and this is what I am matching here
-            best = min(candidates) if len(candidates) else None
+                            nmismatchedtoptaggedjets +=1 #only with respect to the hadronic top where W -> qq, this is what being matched here
+            minCandidate = min(candidateMinimizers) if len(candidateMinimizers) else None
+            best = candidateBranchValues[candidateMinimizers.index(minCandidate)] if len(candidateMinimizers) else None
+            
+            # function for sorting lists
+            #def ExtractIndex(lst):
+                #return [item[1] for item in lst]
+            #def ExtractpT(lst):
+                #return [item[0] for item in lst]
+
+            #TODO should I add if len(QFromWFromH)==2 here? 
```

Comment:
Theoretically yes, but in practice itertools.combinations returns an empty pair so it won't looped on.
It would still be good to have it in the case of H->WW->qqqq, although that one would pass selection only if the two tops are semileptonic and with a mismeasured charge, so practically never I guess.